assets/js/google_map.js

```diff
@@ -116,19 +137,68 @@ async function addMarkers(map, cfgs) {
 
   try {
     for (let cfg of cfgs) {
-      await utils.createMarker(
-        map,
-        cfg,
-        Place,
-        AdvancedMarkerElement,
-        PinElement
+      markers.push(
+        await utils.createMarker(
+          map,
+          cfg,
+          Place,
+          AdvancedMarkerElement,
+          PinElement
+        )
       );
     }
   } catch (e) {
     console.log(e);
   }
 }
 
+async function drawRoute(map, cfgs) {
+  const directionsService = new google.maps.DirectionsService();
+  const directionsRenderer = new google.maps.DirectionsRenderer();
+
+  directionsRenderer.setMap(map);
+
+  if (cfgs.length < 2) {
+    return;
+  }
+
+  const start = new google.maps.LatLng(
+    cfgs[0].position.lat,
+    cfgs[0].position.lng
+  );
+
+  const end = new google.maps.LatLng(
+    cfgs[cfgs.length - 1].position.lat,
+    cfgs[cfgs.length - 1].position.lng
+  );
+
```

Comment:
what are waypoints, things in between right ?

Reply:
Correct.they are a sequence of points where the route should go through.